File: reactapp1.client/src/models/TotalityData.ts
import type { UserPlusTotality } from "./UserPlusTotality";

//Modelo usado no endpoint "/totality" (GET)
export interface TotalityData {
    users: UserPlusTotality[]; //Lista de todas as pessoas
    total_receitas: number; //Total de receitas de todas as pessoas
    total_despesas: number; //Total de despesas de todas as pessoas
    total_saldo: number; //Saldo líquido de todas as pessoas
}

File: ReactApp1.Server/Models/TotalityData.cs
namespace ReactApp1.Server.Models
{
    //Classe que representa os dados de totalidade do endpoint "/Totality" (GET).
    public class TotalityData
    {
        public double total_receitas { get; set; }
        public double total_despesas { get; set; }
        public double total_saldo => total_receitas - total_despesas;
        public IEnumerable<UserPlusTotality> users { get; set; } = [];
    }
}